base/src/Data/Macaw/Analysis/RegisterUse.hs

```diff
@@ -5,22 +5,51 @@ task needed before deleting unused code.
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
 module Data.Macaw.Analysis.RegisterUse
```

Comment:
Module level comments/questions:
- A high-level description of the intent would be useful, or maybe just additional commentary on the `registerUse` function. My interpretation of this module is that it discovers what registers and memory locations must be populated for the input function to have defined behavior (ish).
- From the representation I think I see, this only talks about "top level" register and memory structure - it doesn't say anything about e.g., a pointer to a struct having particular fields initialized
- It seems like this code makes assumptions about calls obeying the ABI - is it the responsibility of the caller to validate that?
- It would be really useful to have a test or two in macaw-x86 to make sure that nobody else breaks anything